### PR TITLE
fix(HitsPerPage): use connector transformItems

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This migrates HitsPerPage to use the transformItems provided by the connector.
It also adds a story to showcase it.

ref #372

**Results**
https://deploy-preview-562--angular-instantsearch.netlify.com/examples/storybook/?path=/story/hitsperpage--with-transformitems

### CODE REVIEW

File: src/hits-per-page/hits-per-page.ts [hunk 1]
```diff
@@ -6,8 +6,14 @@ import { NgAisInstantSearch } from '../instantsearch/instantsearch';
...
-  items: {}[];
-  refine: Function;
+  items: HitsPerPageItem[];
+  refine: (value: number) => void;
+};
+
+export type HitsPerPageItem = {
+  value: number;
+  label: string;
+  isRefined?: boolean;
```
Comment (samouss) [inline src/hits-per-page/hits-per-page.ts]:
```suggestion
  isRefined: boolean;
```

File: src/hits-per-page/hits-per-page.ts [hunk 2]
```diff
@@ -34,19 +40,19 @@ export type HitsPerPageState = {
   `,
 })
 export class NgAisHitsPerPage extends BaseWidget {
+  @Input() public items: HitsPerPageItem[];
```
Comment (samouss) [inline src/hits-per-page/hits-per-page.ts]: The `items` exposed and the `items` provided to the template are two different types.

Comment (tkrugg) [reply src/hits-per-page/hits-per-page.ts]: Good catch, thanks!

Comment (tkrugg) [reply src/hits-per-page/hits-per-page.ts]: fixed here: 2d7239e

File: src/hits-per-page/hits-per-page.ts (hunk 1 above)
Comment (tkrugg) [reply src/hits-per-page/hits-per-page.ts]: fixed here: 2d7239e

**Pull request merged**